common/src/Model/Brush.cpp

```diff
@@ -619,6 +619,24 @@ namespace TrenchBroom {
             face->transform(translationMatrix(delta), lockTexture);
             rebuildGeometry(worldBounds);
         }
+        
+        bool Brush::expand(const BBox3& worldBounds, const FloatType delta, const bool lockTexture) {
+            const NotifyNodeChange nodeChange(this);
+            
+            // move the faces
+            for (BrushFace* face : m_faces) {
+                const Vec3 moveAmount = face->boundary().normal * delta;
+                face->transform(translationMatrix(moveAmount), lockTexture);
+            }
+            
+            // rebuild geometry
+            try {
+                rebuildGeometry(worldBounds);
+                return !m_faces.empty();
+            } catch (GeometryException&) {
+                return false;
```

Comment:
Is the brush broken after this? I usually have a separate `can...` method that tests whether the operation is applicable without changing the brush at all.

Reply:
Yeah, the brush is broken at that point. I added a `canExpand` function.